### PR TITLE
Fix array query string parsing

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

The [query string parser](https://www.npmjs.com/package/querystring) we use at the moment does not support an array as comma-separated values and on top of that the package itself is deprecated and last release was 2 years ago, [querystring node API](https://nodejs.org/api/querystring.html) is considered legacy as well and does not support all parsing requirements we need.

The following request it not correctly parsed at the moment, it result in just one value

```sh
curl 'http://localhost:9596/eth/v1/beacon/states/head/validators?id=0xb89bebc699769726a318c8e9971bd3171297c61aea4a6578a7a4f94b547dcba5bac16a89108b6b6a1fe3695d1a874a0b,0xa99a76ed7796f7be22d5b7e85deeb7c5677e88e511e0b337618f8c4eb61349b4bf2d153f649f7b53359fe8b94a38e44c'
```

To be OpenAPI spec compliant we need to support this query string format, all [other 4 CLs support this format](https://discord.com/channels/593655374469660673/1062235362509996112/1085260177210605679) as well.

The issue was reported by jcrtp from rocketpool, see [discord thread](https://discord.com/channels/593655374469660673/1062235362509996112/1084551528381165568).

**Description**

Updates fastify [querystringParser](https://www.fastify.io/docs/latest/Reference/Server/#querystringparser) to use [qs](https://www.npmjs.com/package/qs) instead of [querystring](https://www.npmjs.com/package/querystring).

`qs` seems to be really popular and is actively maintained. The code looks clean as well and has a good test suite.

We also already use `qs` package in several other places (e.g. [here](https://github.com/ChainSafe/lodestar/blob/a198b7ed1a6328d277ac4dca5483f979b8999b25/packages/api/src/utils/client/format.ts#L8)) in the monorepo.

### CODE REVIEW

File: packages/beacon-node/src/api/rest/base.ts [hunk 1]
```diff
@@ -43,7 +43,15 @@ export class RestApiServer {
     const server = fastify({
       logger: false,
       ajv: {customOptions: {coerceTypes: "array"}},
-      querystringParser: querystring.parse,
+      querystringParser: (str) =>
+        qs.parse(str, {
+          // defaults to 20 but Beacon API spec allows max items of 30
+          arrayLimit: 30,
+          // array as comma-separated values must be supported to be OpenAPI spec compliant
+          comma: true,
```
Comment (nflaig) [inline packages/beacon-node/src/api/rest/base.ts:51]: koa openapi req/res validator uses this library as well and sets [`comma: true`](https://github.com/atlassian/koa-oas3/blob/aae8267a0b7f1cc9e436c449cbc168394ccc85d6/src/config.ts#L92), this is required to be OpenAPI spec compliant, see [Parameter Serialization](https://swagger.io/docs/specification/serialization/)

File: packages/beacon-node/src/api/rest/base.ts [hunk 2]
```diff
@@ -43,7 +43,15 @@ export class RestApiServer {
     const server = fastify({
       logger: false,
       ajv: {customOptions: {coerceTypes: "array"}},
-      querystringParser: querystring.parse,
+      querystringParser: (str) =>
+        qs.parse(str, {
```
Comment (nflaig) [inline packages/beacon-node/src/api/rest/base.ts:47]: see [default parse options](https://github.com/ljharb/qs/blob/9dca37f15de317fe9ad0ced907cdf250ba310880/lib/parse.js#L8)

File: packages/light-client/package.json [hunk 3]
```diff
@@ -77,6 +77,8 @@
   },
   "devDependencies": {
     "@chainsafe/as-sha256": "^0.3.1",
+    "@types/qs": "^6.9.7",
```
Comment (nflaig) [inline packages/light-client/package.json:80]: added as dev deps as we use it in tests `packages/light-client/test/utils/server.ts`

File: packages/beacon-node/src/api/rest/base.ts [hunk 4]
```diff
@@ -43,7 +43,15 @@ export class RestApiServer {
...
+          // defaults to 20 but Beacon API spec allows max items of 30
+          arrayLimit: 30,
```
Comment (nflaig) [inline packages/beacon-node/src/api/rest/base.ts:49]: Beacon API spec defines `30` as `maxItems` for some values, e.g. `getStateValidators` - `id` query, see [spec value](https://github.com/ethereum/beacon-APIs/blob/f087fbf2764e657578a6c29bdf0261b36ee8db1e/apis/beacon/states/validators.yaml#L23)

File: yarn.lock [hunk 5]
```diff
@@ -3406,10 +3406,10 @@
   resolved "https://registry.npmjs.org/@types/prometheus-gc-stats/-/prometheus-gc-stats-0.6.1.tgz"
   integrity sha512-tSj7MRNgEJCSqVRgPL4WIIcslZChgL6gVwnVFsAj2nTx0RdcG7iSpWm+0fkAKEe7e/YQ08TZH4wQWxpFODYqJw==
 
-"@types/qs@^6.9.6":
-  version "6.9.6"
-  resolved "https://registry.npmjs.org/@types/qs/-/qs-6.9.6.tgz"
-  integrity sha512-0/HnwIfW4ki2D8L8c9GVcG5I72s9jP5GSLVF0VIXDW00kmIpA6O33G7a8n59Tmh7Nz0WUC3rSb7PTY/sdW2JzA==
+"@types/qs@^6.9.7":
+  version "6.9.7"
+  resolved "https://registry.yarnpkg.com/@types/qs/-/qs-6.9.7.tgz#63bb7d067db107cc1e457c303bc25d511febf6cb"
```
Comment (nflaig) [inline yarn.lock:3411]: not sure why this was using `registry.npmjs.org` before but `registry.yarnpkg.com` seems correct since we use yarn

**Review by nazarhussain — APPROVED** [review]

**Review by nazarhussain — COMMENTED** [review]
The changes seems ok as it does not show up any regression. I tried to identify what are the changes in the minor release update but seems their [repo](https://github.com/ljharb/qs/tags) does not document it well.

**Comment by nflaig** [comment]
@nazarhussain they document their changes in changelog file, see [CHANGELOG.md](https://github.com/ljharb/qs/blob/main/CHANGELOG.md)

**Review by twoeths — APPROVED** [review]
looks good to me. The reason we didn't catch this issue in unit tests is because we always use our client api there.
probably we should also create unit tests with `curl`

**Review by g11tech — APPROVED** [review]

**Comment by nflaig** [comment]
@tuyennhv yeah, I looked into adding a test for this but there was no quick way to do it. I think for general cases such as this we just need to improve our spec compliance tests

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.6.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.6.0) :tada:

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.7.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.7.0) :tada:

**Pull request merged**